src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -913,6 +914,13 @@ void move(UnixPath source, UnixPath target, CopyOption... options)
             if (!flags.replaceExisting) {
                 throw new FileAlreadyExistsException(
                     target.getPathForExceptionMessage());
+            } else if (!sourceAttrs.isSymbolicLink() || flags.followLinks) {
+                // Ensure source can be moved
+                try {
+                    access(source, W_OK);
```

Comment:
Are you sure W_OK is correct and not R_OK or R_OK|W_OK ?

Reply:
After all I think `R_OK|W_OK` would cover both the atomic and non-atomic (copy + delete) cases.

Reply:
So changed in e226096d15b47bee7cee20ec2994bca93a071433.